internal/pkg/deploy/cloudformation/stack/testdata/pipeline/gh_template.yaml

```diff
@@ -131,6 +131,572 @@ Resources:
+                # Loop through all the artifact buckets created in the stackset
+                Resource:
+                  - !Join [ '', [ 'arn:aws:s3:::', 'fancy-bucket' ] ]
+                  - !Join [ '', [ 'arn:aws:s3:::', 'fancy-bucket', '/*' ] ]
+              - Effect: Allow
+                Action:
+                  # TODO: scope this down if possible
+                  - kms:*
+                # TODO: This might not be necessary. We may only need the KMS key
+                # that is in the same region as the pipeline.
+                # Loop through all the KMS keys used to en/decrypt artifacts
+                # across (cross-regional) pipeline stages, with each stage
+                # backed by a (regional) S3 bucket.
+                Resource:
+                  - arn:aws:kms:us-west-2:1111:key/abcd
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: arn:aws:logs:*:*:*
+              - Effect: Allow
+                Action:
+                  - ecr:GetAuthorizationToken
+                Resource: '*'
+              - Effect: Allow
+                Action:
+                  - ecr:DescribeImageScanFindings
+                  - ecr:GetLifecyclePolicyPreview
+                  - ecr:GetDownloadUrlForLayer
+                  - ecr:BatchGetImage
+                  - ecr:DescribeImages
+                  - ecr:ListTagsForResource
+                  - ecr:BatchCheckLayerAvailability
+                  - ecr:GetLifecyclePolicy
+                  - ecr:GetRepositoryPolicy
+                  - ecr:PutImage
+                  - ecr:InitiateLayerUpload
+                  - ecr:UploadLayerPart
+                  - ecr:CompleteLayerUpload
+                Resource: '*'
+                Condition: { StringEquals: { 'ecr:ResourceTag/copilot-application': phonetool } }
+
+  PretestDeploymentActionpreAction1:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      EncryptionKey: !ImportValue phonetool-ArtifactKey
+      ServiceRole: !GetAtt PretestDeploymentActionpreAction1BuildProjectRole.Arn
+      Artifacts:
+        Type: CODEPIPELINE
+      Cache:
+        Modes:
+          - LOCAL_DOCKER_LAYER_CACHE
+        Type: LOCAL
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        PrivilegedMode: true
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:4.0
+        EnvironmentVariables:
+          - Name: AWS_ACCOUNT_ID
+            Value: !Sub '${AWS::AccountId}'
+          - Name: PARTITION
+            Value: !Ref AWS::Partition
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec: ./buildspec.yml
+      TimeoutInMinutes: 60
+  PretestDeploymentActionpreAction2BuildProjectRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
```

Comment:
is this policy doc the same as the one above? i wonder if it's better to use [`AWS::IAM::POLICY`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-policy.html#cfn-iam-policy-roles) and just list all of the roles ARNs in `Roles` 🤔